File: c/49.c
#include <stdio.h>
#include <math.h>

int prime(int n)
{
	int i;

	if (n % 2 == 0)
		return 0;

	for (i = 3; i < sqrt(n); i += 2)
		if (n % i == 0)
			return 0;

	return 1;
}

int samedig(int a, int b)
{
	int i;
	char as[10] = {0}, bs[10] = {0};

	do ++as[a % 10]; while (a /= 10);
	do ++bs[b % 10]; while (b /= 10);

	for (i = 0; i < 10; ++i)
		if (as[i] != bs[i])
			return 0;

	return 1;
}

int main(void)
{
	int i = 1487, j, k;

	while (++i, j = i + 3330, k = i + 6660)
		if (prime(i) && prime(j) && prime(k) && samedig(i, j) && samedig(j, k))
			break;

	printf("%d%d%d\n", i, j, k);
	return 0;
}

File: hs/49.hs
import Data.List (find, sort)
import Data.Maybe (fromJust)

fs = floor . sqrt . fromIntegral

prime n | n == 2    = True
        | even n    = False
        | otherwise = notElem 0 $ map (\m -> mod n m) [3,5..fs n]

samedigs a b = (sort $ show a) == (sort $ show b)

solution n = prime n && prime o && prime p && samedigs n o && samedigs o p
    where o = n + 3330
          p = n + 6660

main = do
    putStr $ show sol
    putStr $ show $ sol + 3330
    putStr $ show $ sol + 6660
    where sol = fromJust $ find solution [1488..]
